liquibase-extension-testing/src/main/java/liquibase/extension/testing/setup/TestSetupEnvironment.java

```diff
@@ -0,0 +1,35 @@
+package liquibase.extension.testing.setup;
+
+import liquibase.extension.testing.testsystem.DatabaseTestSystem;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+public class TestSetupEnvironment {
+
+    public final String username;
+    public final String url;
+    public final String password;
+    public final Connection connection;
+
+    public final Connection altConnection;
+    public final String altUsername;
+    public final String altUrl;
+    public final String altPassword;
+
+    public String errorMessage;
+
+    public TestSetupEnvironment(DatabaseTestSystem testSystem, DatabaseTestSystem altSystem) throws SQLException {
+        this.connection = testSystem.getConnection();
+        this.url = testSystem.getConnectionUrl();
+        this.username = testSystem.getUsername();
+        this.password = testSystem.getPassword();
+
+        this.altUrl = this.url.replace(testSystem.getCatalog(), testSystem.getAltCatalog());
+        this.altUsername = this.username;
+        this.altPassword = this.password;
+        this.altConnection = DriverManager.getConnection(this.altUrl, this.altUsername, this.altPassword);
```

Comment:
Why are you getting the connection differently for the altSystem? If this is intentional, a comment here explaining why would be useful to future generations of Liquibase maintainers.

Reply:
This is for the existing CommandTest stuff. The way the tests are set up, they need to be able to have an "alt system" for things like diff commands which need to work against two separate connections.

What we have in this TestSetupEnvironment was made for the CommandTests, but with an eye toward being part of a generic way of managing the system. I'm planning on trying to refactor this code somehow, but I'm not quite sure how yet. I want to see how the tests we write over the next few weeks can leverage it and what they need.

So I figure this code here is pretty temporary, and just there as a shim for the existing CommandTests until we can do it better soon.